### DISCUSSION TITLE
Error rates in the log file and cer.png don't match

### DISCUSSION THREAD

**Discussion by mitali-a** [espnet/espnet#4758] [Q&A]
@kan-bayashi 
I tried running the exact recipe for an4 on my pc and I got similar results for CER and SER for mtlalpha=1.0(ESPnet result: cer=15.1% on dev set and cer=8.8% on test set). 

Espnet result:
4. CTC only using word-LM

exp/train_nodev_pytorch_train_mtlalpha1.0/decode_train_dev_decode_ctcweight1.0_rnnlm1.0_lm_word100/result.txt:|   SPKR      |   # Snt     # Wrd    |   Corr         Sub        Del         Ins         Err      S.Err    |
exp/train_nodev_pytorch_train_mtlalpha1.0/decode_train_dev_decode_ctcweight1.0_rnnlm1.0_lm_word100/result.txt:|   Sum/Avg   |    100       1915    |   86.8         5.7        7.5         1.9        15.1       69.0    |
exp/train_nodev_pytorch_train_mtlalpha1.0/decode_test_decode_ctcweight1.0_rnnlm1.0_lm_word100/result.txt:|   SPKR     |   # Snt     # Wrd   |   Corr        Sub         Del        Ins        Err      S.Err   |
exp/train_nodev_pytorch_train_mtlalpha1.0/decode_test_decode_ctcweight1.0_rnnlm1.0_lm_word100/result.txt:|   Sum/Avg  |    130       2565   |   92.6        3.5         3.8        1.5        8.8       56.2   |

My results:
write a CER (or TER) result in exp/train_nodev_pytorch_train_mtlalpha1.0/decode_train_dev_decode_ctcweight1.0_lm_word100/result.txt
|   SPKR     |  # Snt     # Wrd   |   Corr       Sub        Del        Ins       Err      S.Err   |
|   Sum/Avg  |   100       1915   |   85.4       5.9        8.7        1.5      16.1       66.0   |
write a CER (or TER) result in exp/train_nodev_pytorch_train_mtlalpha1.0/decode_test_decode_ctcweight1.0_lm_word100/result.txt
|  SPKR     |  # Snt    # Wrd   |  Corr        Sub       Del       Ins       Err     S.Err   |
|  Sum/Avg  |   130      2565   |  91.6        4.8       3.6       0.7       9.1      53.1   |

Now, when I checked the cer.png file, cer never falls below 20% for validation set during entire training and still during decoding stage, final cer on dev set is around 16%. Even the log file error rates and loss values don't match with the ones in cer.png and loss.png for all the epochs mentioned(for eg. Epoch 10). My concern is how is the best/final model selected and why there is this mismatch in these files.
[log.pdf](https://github.com/espnet/espnet/files/9959068/log.pdf)
![cer](https://user-images.githubusercontent.com/55866065/200511926-09cd7ca9-a3b7-449f-a25d-8384996cd971.png)
![loss](https://user-images.githubusercontent.com/55866065/200511931-3a2f3a1b-61ae-4c94-98af-64df6cf21866.png)
